### PR TITLE
V0.2.0 review updates

### LINKED ISSUE #44: Refresh README
The currently README is out of date and should be updated to give an intro into how to navigate this repo. Namely, it should provide:

1. A basic README structure since the bulk of the content will be hosted in gh-pages
2. A [link](https://rog-golang-buddies.github.io/golang-template-repository/) to this repo's gh-pages site so that users are aware of how to get started.

### PR DESCRIPTION
**Pull request by haani-niyaz**
### Docs update

1. Readme has been rewritten to access starter docs
2. Add sample README template
3. Fixed pre-commit docs to capture pre-commit ci integration
4. Follow a consistent naming convention for workflow names and steps
5. Use a lowercase file naming convention for workflows (It doesn't have to be lowercase but I picked this to keep all of them consistent)
6. Add GoReleaser docs 

Closes #44

### Limit builds

- Removed  gh action `on:pull_request` since `on:push` results are available in the PR and this will avoid duplicate runs

Addresses #43

### Dockerfile enhancements

Closes #45 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,64 @@
...
+
+_You can also document commands to lint the code or run tests. These steps help to ensure high code quality and reduce the likelihood that the changes inadvertently break something. Having instructions for running tests is especially helpful if it requires external setup, such as starting a Selenium server for testing in a browser._
+
+## Authors and acknowledgment
+
+_Show your appreciation to those who have contributed to the project._
+
+## License
+
+_For open source projects, say how it is licensed._
```
Comment (mridulganga) [inline README.md:64]: we can probably suggest 

Apache 2.0 or MIT here

File: docs/continuous-integration/pre-commit.md [hunk 2]
```diff
@@ -23,7 +23,9 @@ Any detected failures will abort the commit.
 
     No configuration required.
 
-All hooks in `.pre-commit-config.yaml` will be executed via the pre-commit Github workflow in `.github/workflows/pre-commit.yml`.
+All hooks in `.pre-commit-config.yaml` will be executed via the automatic integration provided from [pre-commit ci](https://pre-commit.ci/).
+
+`pre-commit ci` runs automatically on pull request commits.
```
Comment (mridulganga) [inline docs/continuous-integration/pre-commit.md:28]: should we mention that pre-commit may make changes in the PR and the dev might need to pull the changes before creating more commits?

File: .github/workflows/gitleaks.yml [hunk 3]
```diff
@@ -1,26 +1,22 @@
-name: Gitleaks
-
-on: [pull_request, push, workflow_dispatch]
-
+name: gitleaks
+on: [push]
```
[the remark this answers is not in the document]
Comment (haani-niyaz) [reply .github/workflows/gitleaks.yml:2]: I considered this but we sacrifice getting early feedback if we don't do it within the branches.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -1,28 +1,29 @@
...
-  tag:
+  semantic-release:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         node-version:
           - 16.x
     steps:
-      - name: Checkout
+      -
```
[the remark this answers is not in the document]
Comment (haani-niyaz) [reply .github/workflows/release.yml:14]: I'm going to leave it as a separate task as I haven't change it from what it was. We can always do it later.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -1,28 +1,29 @@
...
+        name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
-      - name: Release
+      -
```
[the remark this answers is not in the document]
Comment (haani-niyaz) [reply .github/workflows/release.yml:19]: I'm going to leave it as a separate task as I haven't change it from what it was. We can always do it later.

File: docs/continuous-integration/pre-commit.md (hunk 2 above)
Comment (haani-niyaz) [reply docs/continuous-integration/pre-commit.md:28]: yup good point 👍

File: .github/workflows/mkdocs.yml [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
         with:
           python-version: 3.x
-      - run: pip install mkdocs-material
-      - run: mkdocs gh-deploy --force
+      - name: Install mkdocs-material
+        run: pip install mkdocs-material
```
[the remark this answers is not in the document]
Comment (shukra-in-spirit) [reply .github/workflows/mkdocs.yml:17]: Can you make a fresh issue for this @mridulganga, we went for anything that we were able to get working.

Now, would be a good time to start working on evaluating and taking out non-golang bits.

File: README.md (hunk 1 above)
Comment (shukra-in-spirit) [reply README.md:64]: I added this @haani-niyaz sorry for high-jacking your PR.

**Comment by shukra-in-spirit** [comment]
Thanks, @haani-niyaz and @mridulganga for the help.
@haani-niyaz have you added the comment to trigger the semantic releaser and make it v0.2.0?
I saw that none of the other commits bumped the bug-fix versions as expected. Maybe, there is a need to encourage members to follow the commit message rules to trigger releases accordingly,

**Review by shukra-in-spirit — APPROVED** [review]
Please check above comment before merging @haani-niyaz.

**Comment by haani-niyaz** [comment]
> Thanks, @haani-niyaz and @mridulganga for the help. @haani-niyaz have you added the comment to trigger the semantic releaser and make it v0.2.0? I am approving the PR, can you merge it if the comment message is apt. I saw that none of the other commits bumped the bug-fix versions as expected. Maybe, there is a need to encourage members to follow the commit message rules to trigger releases accordingly,

@shukra-in-spirit, I was planning on merging #45 to this PR as it completes the `v0.2.0` release requirements. Once done, I'll bump this with a `feat` commit to trigger `v0.2.0`

> I saw that none of the other commits bumped the bug-fix versions as expected.

Do you mean with the `fix` type in the commit message or something else?

I also think some release planning is in order to keep everyone on the same page.

**Comment by haani-niyaz** [comment]
> Yeap. With a fix mention in the comment to bump up the x in v0.1.x

@shukra-in-spirit, I don't see any commits with `fix` type in the `main` branch's commit log. Possibly squashed when merging.

**Pull request merged**